x-pack/plugins/ml/public/application/datavisualizer/index_based/index_data_visualizer.tsx

```diff
@@ -86,8 +100,8 @@ export const IndexDataVisualizerPage: FC = () => {
             return false;
           }
         },
-        dataTestSubj: 'dataVisualizerCreateAdvancedJobCard',
-      },
+        'data-test-subj': 'dataVisualizerCreateAdvancedJobCard',
+      } as ResultLink,
```

Comment:
Rather than asserting as `ResultLink`, this function could have a return signature of `Promise<ResultLink[]>`

Reply:
Updated here https://github.com/elastic/kibana/pull/131342/commits/846bd87fbe77a488d47e1a9ee7d2d56a17011e74